gneiss/regression/tests/test_regression.py

```diff
@@ -0,0 +1,118 @@
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.],
+                                  [0., 0.]],
+                                 index=['s1', 's2', 's3', 's4', 's5'],
+                                 columns=['Y1', 'Y2'])
+        pdt.assert_frame_equal(exp_resid, res.residuals())
+
+    def test_ols_cli(self):
+        # TODO: Is there a better way to test this?
+        in_table = get_data_path("test_ols_composition.qza")
+        in_tree = get_data_path("test_tree.qza")
+        in_metadata = get_data_path("test_ols_metadata.txt")
+
+        cmd = ("qiime gneiss ols-regression "
+               "--i-table %s "
+               "--i-tree %s "
+               "--o-linear-model test_ols "
+               "--p-formula 'ph' "
+               "--m-metadata-file %s")
+        proc = subprocess.Popen(cmd % (in_table, in_tree, in_metadata),
+                                shell=True)
+        proc.wait()
+
+        self.assertTrue(os.path.exists("test_ols.qza"))
+
+        os.remove("test_ols.qza")
+
+
+class TestMixedLMPlugin(TestMixedLM):
+
+    def test_mixedlm_balances(self):
+
+        res = lme_regression(formula="x1 + x2", table=self.table,
+                             metadata=Metadata(self.metadata), tree=self.tree,
+                             groups="groups")
+        res.fit()
+        exp_pvalues = pd.DataFrame(
+            [[4.82688604e-236,  4.4193804e-05,  3.972325e-35,  3.568599e-30],
+             [0.0994110906314,  4.4193804e-05,  3.972325e-35,  3.568599e-30]],
+            index=['Y1', 'Y2'],
+            columns=['Intercept', 'groups RE', 'x1', 'x2'])
+
+        pdt.assert_frame_equal(res.pvalues, exp_pvalues,
+                               check_less_precise=True)
+
+        exp_coefficients = pd.DataFrame(
+            [[4.211451,  0.0935786, 1.022008, 0.924873],
+             [0.211451,  0.0935786, 1.022008, 0.924873]],
+            columns=['Intercept', 'groups RE', 'x1', 'x2'],
+            index=['Y1', 'Y2'])
+
+        pdt.assert_frame_equal(res.coefficients(), exp_coefficients,
+                               check_less_precise=True)
+
+    def test_lme_cli(self):
+        # TODO: Is there a better way to test this?
+        in_table = get_data_path("test_lme_composition.qza")
+        in_tree = get_data_path("test_tree.qza")
+        in_metadata = get_data_path("test_lme_metadata.txt")
+
+        cmd = ("qiime gneiss lme-regression "
+               "--i-table %s "
+               "--i-tree %s "
+               "--o-linear-mixed-effects-model test_lme "
+               "--p-formula 'ph' "
+               "--p-groups 'host_subject_id' "
+               "--m-metadata-file %s")
+        proc = subprocess.Popen(cmd % (in_table, in_tree, in_metadata),
+                                shell=True)
+        proc.wait()
+
+        self.assertTrue(os.path.exists("test_lme.qza"))
+        os.remove("test_lme.qza")
+
```

Comment:
This has been addressed using the Artifact API.  Really want to avoid using tar here.